go/vt/vtgate/grpcvtgateservice/server.go

```diff
@@ -192,19 +195,22 @@ func (vtg *VTGate) StreamExecute(request *vtgatepb.StreamExecuteRequest, stream
 		})
 	})
 
-	// even if there is an error, session could have been modified.
-	// So, this needs to be sent back to the client. Session is sent in the last stream response.
-	lastErr := stream.Send(&vtgatepb.StreamExecuteResponse{
-		Session: session,
-	})
-
 	var errs []error
 	if vtgErr != nil {
 		errs = append(errs, vtgErr)
 	}
-	if lastErr != nil {
-		errs = append(errs, lastErr)
+
+	if sendSessionInStreaming {
+		// even if there is an error, session could have been modified.
+		// So, this needs to be sent back to the client. Session is sent in the last stream response.
+		lastErr := stream.Send(&vtgatepb.StreamExecuteResponse{
+			Session: session,
+		})
+		if lastErr != nil {
+			errs = append(errs, lastErr)
+		}
```

Comment:
These changes were previously not behind a flag?

Reply:
If this is already there in a released version of Vitess, then maybe we should keep the flag to default false in this release too.

Reply:
I feel the breaking change was in v17 so, if you are already on v17 then the users would have already fixed it. 
I do not see a point in keeping it to `false` in v18.

Reply:
Yeah, but if someone upgrades from an existing version of v17 which is already released, its going to be a breaking change for them. We do not ask users to do major upgrades only with the latest patch versions.

Reply:
The grpc users are very limited and none except one has reported it, so for others, it should be a no-op.
It covers the functionality provided in v17 only i.e. transactions in streaming for grpc api.

Reply:
But it is still going to be a breaking change from v17.0.0 to v18.0.0. We should avoid it because it is easily avoidable by just postponing when we deprecate and delete the flag.